beacon/beacon.go

```diff
@@ -922,7 +937,15 @@ func buildSignedProposal(ctx context.Context, signer signing.Signer, epoch types
 	return signature
 }
 
-func signMessage(signer signing.Signer, message interface{}, logger log.Log) []byte {
+func signFirstVotingMessage(signer signing.Signer, message *FirstVotingMessageBody, logger log.Log) []byte {
```

Comment:
what's the diff btwn signFirstVotingMessage and signFollowingVotingMessage?

Reply:
they accept different message types. Scale codec doesn't work with `interface{}`. Alternatively we can move `codec.Encode` to the caller side but then only `signer.Sign` is left and no point in a function :)

Reply:
right. in that case maybe just get rid of signXXXMessage all together?

Reply:
and done

Reply:
you could replace interface{} with codec.Encodable (which should be an alias for scale.Encodable)